src/analysis/buildTypeWeights.ts

```diff
@@ -30,6 +63,145 @@ function buildTypeWeightsFromSchema(
     schema: GraphQLSchema,
     typeWeightsConfig: TypeWeightConfig = defaultTypeWeightsConfig
 ): TypeWeightObject {
-    throw Error(`getTypeWeightsFromSchema is not implemented.`);
+    if (!schema) throw new Error('Must provide schema');
+
+    //  Merge the provided type weights with the default to account for missing values
+    const typeWeights: TypeWeightConfig = {
+        ...defaultTypeWeightsConfig,
+        ...typeWeightsConfig,
+    };
+
+    // Confirm that any custom weights are positive
+    Object.entries(typeWeights).forEach((value: [string, number]) => {
+        if (value[1] < 0) {
+            throw new Error(`Type weights cannot be negative. Received: ${value[0]}: ${value[1]} `);
+        }
+    });
+
+    const result: TypeWeightObject = {};
+
+    const typeMap: ObjMap<GraphQLNamedType> = schema.getTypeMap();
+
+    // Handle Object, Interface, Enum and Union types
+    Object.keys(typeMap).forEach((type) => {
+        const currentType: GraphQLNamedType = typeMap[type];
+        // Get all types that aren't Query or Mutation or a built in type that starts with '__'
+        if (
+            currentType.name !== 'Query' &&
+            currentType.name !== 'Mutation' &&
+            !currentType.name.startsWith('__')
+        ) {
+            if (isObjectType(currentType) || isInterfaceType(currentType)) {
+                // Add the type to the result
+                result[type] = {
```

Comment:
Can we make this all lower case? The queries come in all lower case. It is probably easier to lowercase everything here than uppercase only the first letter in the complexity analysis.